### PR TITLE
convert settings.gradle to kotlin

### PR DESCRIPTION
**Pull request by soloturn**
convert settings.gradle to kotlin

### CODE REVIEW

**Comment by Cervator** [comment]
Seems straight forward enough, although I wonder is there any particular reasoning / added benefit to the change? I know sometimes Groovy stuff can be slow on its own.

**Comment by soloturn** [comment]
> Seems straight forward enough, although I wonder is there any particular reasoning / added benefit to the change? I know sometimes Groovy stuff can be slow on its own.

gradle switched the default:
https://blog.gradle.org/kotlin-dsl-is-now-the-default-for-new-gradle-builds

**Comment by soloturn** [comment]
how do you normally get a review, a feedback or a merge, @Cervator @jdrueckert @BenjaminAmos ?

**Comment by jdrueckert** [comment]
> how do you normally get a review, a feedback or a merge, @Cervator @jdrueckert @BenjaminAmos ?

Easiest way to get attention is via discord. We do have a review-hotlist channel there. I personally also use the activity-feed channel which is synced with some repos and automatically shows new PRs, commits and comments.

Admittedly sometimes I see new changes but don't have time in that moment and then they slip my mind. Probably similar for others as well due to day job and other real life obligations.

It also frequently happens that people comment to ask something but when the answer comes don't approve. Or approve but don't merge. Not entirely sure why that is though.

File: settings.gradle.kts
Comment (jdrueckert) [inline settings.gradle.kts]:
```suggestion
        possibleSubprojectDir.walkTopDown().forEach { it.listFiles { file -> 
            file.isFile && file.name == "subprojects.settings.gradle" }?.forEach { subprojectsSpecificationScript ->
```

File: settings.gradle.kts
Comment (jdrueckert) [inline settings.gradle.kts]:
```suggestion
}

```

**Review by jdrueckert — APPROVED** [review]

**Comment by jdrueckert** [comment]
@soloturn From my pov this can be merged once the checks are green again.
If I forget to check on this today again, apologies, please do ping me again for merging in that case.

**Pull request merged**
